lpc845/scripts/update.sh

```diff
@@ -0,0 +1,18 @@
+#!/usr/bin/env bash
+
+cp LPC845.xml LPC845.svd
+
+for patch in patches/*
+do
+	patch \
+		--silent \
+		--force \
+		--reject-file=- \
+		LPC845.svd $patch
+done
+
+rm -rf src && mkdir src &&
+svd2rust -i LPC845.svd &&
+form -i lib.rs -o src && rm lib.rs &&
+cargo fmt &&
+rustfmt build.rs
```

Comment:
I strongly recommend against using this approach. The only reason this exists is because I can't legally add the LPC82x file I'm using to this repository.

Instead, I'd recommend the following:
1. Add the SVD file to the repository (you already did this)
2. Fix the SVD file by making commits in this repository (I recommend one commit per fix).
3. Integrate it into the build script in the top-level directory.

Reply:
I prefer to keep it more obvious what's the nxp stuff & what we changed.
What are the reasons against it?

What do you think about adopting the (stm32-rs)[https://github.com/stm32-rs/stm32-rs] structure (haven't looked that closely yet)?

Reply:
I found dealing with the patch files not very convenient in the past, but it's really not that bad. I've been thinking that using Git would make it easier to compare and merge new versions, but I might be wrong.

@dbrgn Thoughts?

Reply:
@david-sawatzke 
> What do you think about adopting the stm32-rs structure (haven't looked that closely yet)?

Sorry, I missed this in my previous reply.

I don't know. I know that stm32-rs uses its own format to patch files with some Python code. I think they also make sure that identical peripherals across devices use the same types, maybe?

Overall, what they do is much more advanced than what we do here. I think it would be a great idea to look into it in detail and see if that can be adopted or copied for our use. I won't be able to do hat any time soon though.

If you are (or anyone else is) interested in looking into this and reporting back here, that would certainly be valuable.